sdk/src/signer/keypair.rs

```diff
@@ -113,6 +113,46 @@ where
     }
 }
 
+impl EncodableKey for Keypair {
+    fn pubkey_string(&self) -> Result<String, Box<dyn error::Error>> {
+        Ok(self.pubkey().to_string())
+    }
+
+    fn read_key<R: Read>(reader: &mut R) -> Result<Self, Box<dyn error::Error>> {
+        read_keypair(reader)
+    }
+
+    fn read_key_file<F: AsRef<Path>>(path: F) -> Result<Self, Box<dyn error::Error>> {
```

Comment:
Yep, that is a good idea. I will make those changes.

For the deprecations, how about we do it in a separate PR (so deprecate since `1.16`?)? It would be nice if this current PR gets merged in soon and I am afraid that the deprecations will break CI.

Reply:
yeah fine to add the deprecation attribute separately